2020/4/passport/src/main.rs

```diff
@@ -0,0 +1,248 @@
+                                    Some(pid) => is_valid_passport_id(pid),
+                                    None => false,
+                                },
+                                None => false,
+                            },
+                            None => false,
+                        },
+                        None => false,
+                    },
+                    None => false,
+                },
+                None => false,
+            },
+            None => false,
+        }
+    }
+}
+
+fn is_valid_year(value: &str, min: usize, max: usize) -> bool {
+    if value.len() != 4 {
+        return false;
+    }
+
+    let v = value.parse::<usize>().unwrap();
+    min <= v && v <= max
+}
+
+fn is_valid_hair_color(value: &str) -> bool {
+    if value.len() != 7 {
+        return false;
+    }
+
+    let mut chars = value.chars();
+    if chars.next().unwrap() != '#' {
+        return false;
+    }
+
+    for _ in 1..6 {
+        if !chars.next().unwrap().is_digit(16) {
+            return false;
+        }
+    }
+    true
+}
+
+fn is_valid_eye_color(value: &str) -> bool {
+    match value {
+        "amb" => true,
+        "blu" => true,
+        "brn" => true,
+        "gry" => true,
+        "grn" => true,
+        "hzl" => true,
+        "oth" => true,
+        _ => false,
+    }
+}
+
+fn is_valid_passport_id(value: &str) -> bool {
+    value.len() == 9 && value.chars().all(|c| c.is_digit(10))
+}
+
+fn is_valid_height(value: &str) -> bool {
+    if value.ends_with(&"cm") {
+        return is_between(value, 150, 193);
+    } else if value.ends_with(&"in") {
+        return is_between(value, 59, 76);
+    }
+
+    false
+}
+
+fn is_between(value: &str, min: usize, max: usize) -> bool {
+    let len = value.len();
+    let v = value[0..len - 2].parse::<usize>();
+    match v {
+        Ok(vv) => vv >= min && vv <= max,
+        Err(_) => false,
+    }
+}
+
+#[test]
+fn is_valid_passport_first() {
+    let passport = Passport::from_string(
+        "ecl:gry pid:860033327 eyr:2020 hcl:#fffffd
+        byr:1937 iyr:2017 cid:147 hgt:183cm",
+    );
+
+    assert_eq!(true, passport.is_valid());
+}
+
+#[test]
+fn is_valid_passport_second() {
+    let passport = Passport::from_string(
+        "iyr:2013 ecl:amb cid:350 eyr:2023 pid:028048884
+        hcl:#cfa07d byr:1929",
+    );
+
+    assert_eq!(false, passport.is_valid());
+}
+
+#[test]
+fn is_valid_passport_third() {
+    let passport = Passport::from_string(
+        "hcl:#ae17e1 iyr:2013
+        eyr:2024
+        ecl:brn pid:760753108 byr:1931
+        hgt:179cm",
+    );
+
+    assert_eq!(true, passport.is_valid());
+}
+
+#[test]
+fn is_valid_passport_fourth() {
+    let passport = Passport::from_string(
+        "hcl:#cfa07d eyr:2025 pid:166559648
+        iyr:2011 ecl:brn hgt:59in",
+    );
+
+    assert_eq!(false, passport.is_valid());
+}
+
+#[test]
+fn test_height_60in() {
+    assert_eq!(true, is_valid_height("60in"));
```

Comment:
I believe you can do:

```suggestion
    assert!(is_valid_height("60in"));
```